python/ray/remote_function.py

```diff
@@ -120,6 +120,21 @@ def __init__(
         if "runtime_env" in self._default_options:
             self._default_options["runtime_env"] = self._runtime_env
 
+        # Pre-calculate runtime env info, to avoid re-calculation at `remote`
+        # invocation. When `remote` call has specified extra `option` field,
+        # runtime env will be merged and re-serialized.
+        #
+        # Caveat: for `func.option().remote()`, we have to recalculate serialized
+        # runtime env info upon every call. But it's acceptable since pre-calculation
```

Comment:
to be more clear,
```
To support dynamic runtime envs in `func.option(runtime_env={...}).remote()`, we recalculate the serialized runtime env info in the `option` call. If there are multiple calls to a same option, one can save the calculation by `opt_f = func.option(runtime_env={...}); [opt_f.remote() for i in range(many)]`.
```

Reply:
I'm not sure I follow the "If there are multiple calls to a same option" part, since we don't do any caching for `option` calls.

Reply:
Adopted other comments.